### PR TITLE
feat(core): configure Stack SNS notification ARNs on the Stack construct

### PR DESCRIPTION
**Pull request by comcalvi**
### Issue # (if applicable)

#8581.

### Reason for this change

It is easier and clearer to specify the SNS Topic ARNs on the stack construct itself instead of passing it as a command line argument.

### Description of changes

Added a new optional stack prop, `notificationArns`, that is written to the CloudAssembly and concatenated with the CLI option `--notification-arns`. 

Don't forget to select stacks by hierarchical ID (currently display name, in our tests) when writing certain test code. Otherwise, the tests may not select the stack you expect.

Depends on: https://github.com/cdklabs/cdk-assets/pull/87 and https://github.com/cdklabs/cloud-assembly-schema/pull/58.

### Description of how you validated changes

Unit tests + CLI integ test. Framework integ tests not included because they would require an externally-created SNS Topic, which is not what we want in integ tests; besides, the case is covered by the CLI integ test. 

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/core/lib/stack.ts [hunk 1]
```diff
@@ -450,6 +464,7 @@ export class Stack extends Construct implements ITaggable {
       throw new Error(`Stack name must be <= 128 characters. Stack name: '${this._stackName}'`);
     }
     this.tags = new TagManager(TagType.KEY_VALUE, 'aws:cdk:stack', props.tags);
+    this._notificationArns = props.notificationArns ?? [];
```
Comment (rix0rrr) [inline packages/aws-cdk-lib/core/lib/stack.ts:467]: We should probably validate that these strings don't contain any tokens.

**Review by rix0rrr — COMMENTED** [review]
I don't necessarily have a problem with this, but I'm a bit dubious on the practical use cases.

**Review by rix0rrr — CHANGES_REQUESTED** [review]
See previous review

**Comment by comcalvi** [comment]
@rix0rrr could you review: https://github.com/cdklabs/cloud-assembly-schema/pull/29? That needs to merge before this PR can merge.

**Review by rix0rrr — APPROVED** [review]

**Pull request merged**